modules/pool/lib/pool-gql-loader.service.ts

```diff
@@ -77,11 +79,46 @@ export class PoolGqlLoaderService {
             throw new Error('Pool exists, but has an unknown type');
         }
 
-        return this.mapPoolToGqlPool(
+        const mappedPool = this.mapPoolToGqlPool(
             pool,
             pool.userWalletBalances,
             userAddress ? pool.staking.map((staking) => staking.userStakedBalances).flat() : [],
         );
+
+        // load rate provider data into PoolTokenDetail model
+        for (const token of mappedPool.poolTokens) {
+            if (token.priceRateProvider && token.priceRateProvider !== ZERO_ADDRESS) {
+                const rateprovider = await prisma.prismaPriceRateProviderData.findUniqueOrThrow({
+                    where: {
+                        chain_rateProviderAddress: { chain: chain, rateProviderAddress: token.priceRateProvider },
+                    },
+                });
+                token.priceRateProviderData = {
+                    ...rateprovider,
+                    address: rateprovider.rateProviderAddress,
+                };
```

Comment:
so you decided to return it on pool query? we should start to look into requested fields to limit DB load when it's not needed.

Reply:
agree